### PR TITLE
Link encounter with task

### LINKED ISSUE #92: Link encounter with a task id

### PR DESCRIPTION
**Pull request by BuchiNy**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -459,12 +459,13 @@ constructor(
...
-          fhirCarePlanGenerator.completeTask(
-            backReference.asReference(ResourceType.Task).extractId(),
-            it.status,
-          )
+      extras.forEach { resource ->
+        if (resource is Encounter) {
+          val taskReference = backReference.asReference(ResourceType.Task)
+          val taskID = taskReference.extractId()
+          val encounterStatus = resource.status
+          val encounterReference = resource.logicalId.removePrefix("#")
```
Comment (sevenreup) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:467]: This should be converted into a reference, you can call asReference like how we did with the task

**Review by sevenreup — CHANGES_REQUESTED** [review]

**Comment by sevenreup** [comment]
@BuchiNy test the PR again with the changes that Comfort made, and just comment with Task ids that have the new reference set

**Comment by BuchiNy** [comment]
@sevenreup : Task/ab8d9b68-0276-458a-a2ec-df134c07f7fc

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt [hunk 2]
```diff
@@ -459,12 +459,13 @@ constructor(
...
-      extras.forEach {
-        if (it is Encounter) {
-          fhirCarePlanGenerator.completeTask(
-            backReference.asReference(ResourceType.Task).extractId(),
-            it.status,
-          )
+      extras.forEach { resource ->
+        if (resource is Encounter) {
+          val taskReference = backReference.asReference(ResourceType.Task).extractId()
+          val taskID = taskReference
```
Comment (sevenreup) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt]: This can be one line, why are we creating a taskReference and the taskID?

**Review by sevenreup — CHANGES_REQUESTED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt (hunk 2 above)
Comment (BuchiNy) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt]: alright noted

Comment (BuchiNy) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt]: @sevenreup changes made

**Review by sevenreup — APPROVED** [review]
